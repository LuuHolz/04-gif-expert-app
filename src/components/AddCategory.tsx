import { useState } from "react";

type AddCategoryProps = {
  onNewCategory: (category: string) => void;
};

const AddCategory = ({ onNewCategory }: AddCategoryProps) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }: { target: HTMLInputElement }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

export { AddCategory };
