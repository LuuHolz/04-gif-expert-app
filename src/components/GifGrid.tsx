import { GifItem as GifGridItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

const GifGrid = ({ category }: { category: string | number }) => {
  const { images, isLoading } = useFetchGifs(category as string);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}

      <div className="card-grid">
        {images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

export { GifGrid };
