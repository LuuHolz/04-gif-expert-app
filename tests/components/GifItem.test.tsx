import { render, screen } from "@testing-library/react";
import { GifItem } from '../../src/components/GifItem';


describe('Pruebas en GifItem', () => {

    const title = 'Bithday';
    const url = 'http://birthday.com/Birthday.jpg';

    test('Debe de hacer match con el snapshot', () => {

        const { container } = render( <GifItem title={ title } url={ url }/>);
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagencon el URL y el ALT indicado', () => {
        render( <GifItem title={ title } url={ url }/>);

        const img = screen.getByRole('img') as HTMLImageElement;
        const { src, alt } = img;
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    })

    test('Debe de mostrar el titulo en el componente', () => {
        render( <GifItem title={ title } url={ url }/>);

        expect( screen.getByText( title ) ).toBeTruthy();
    })
    
    

});