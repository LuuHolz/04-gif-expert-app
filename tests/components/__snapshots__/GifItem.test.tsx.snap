// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Bithday"
      src="http://birthday.com/Birthday.jpg"
    />
    <p>
      Bithday
    </p>
  </div>
</div>
`;
